aip/auth/4110.md

```diff
@@ -0,0 +1,153 @@
+---
+aip:
+  id: 4110
+  scope: auth
+  state: approved
+  created: 2018-06-24
+  updated: 2019-06-24
+permalink: /auth/4110
+redirect_from:
+  - /4110
+---
+
+# Application Default Credentials
+
+Google auth libraries use a strategy called _Application Default Credentials
+(ADC)_ to detect and select credentials based on environment or context. With 
+ADC, developers should be able to run the code in different environments and the
+supporting systems fetch the appropriate credentials based on each environment
+in an effortless manner.
+
+Auth libraries following the standards in these AIPs are known as _"Google
+Unified Auth Clients"_, or _GUAC_ for short. The resulting libraries are
+colloquially called _GUACs_.
+
+**Note:** Because this AIP describes guidance and requirements in a
+language-neutral way, it uses generic terminology which may be imprecise or
+inappropriate in certain languages or environments.
+
+## Guidance
+
+### Credential Types
+
+This section outlines the supported credential types of the ADC.
+
+#### GCloud CLI Credential
```

Comment:
Please only go down to level three. The infrastructure does not know what to do with level 4+ and we may end up using them for other purposes.

Context: https://google.aip.dev/8#document-structure (note at the bottom).